src/mpyl/stages/discovery.py

```diff
@@ -6,25 +6,43 @@
 import logging
 import os
 import pickle
-from dataclasses import dataclass
+import subprocess
 from pathlib import Path
 from typing import Optional
 
-from ..constants import RUN_ARTIFACTS_FOLDER
+from ..constants import RUN_ARTIFACTS_FOLDER, ROOT_PATH
 from ..project import Project
 from ..project import Stage
 from ..project_execution import ProjectExecution
 from ..run_plan import RunPlan
 from ..steps import deploy
 from ..steps.collection import StepsCollection
 from ..steps.models import Output, ArtifactType
-from ..utilities.repo import Changeset, Repository
-
-
-@dataclass(frozen=True)
-class DeploySet:
-    all_projects: set[Project]
-    projects_to_deploy: set[Project]
+from ..utilities.repo import Changeset
+
+
+def find_projects() -> list[Path]:
+    # Ideally we'd use globbing here, as it's super easy to read:
+    #   projects = root.rglob(f"deployment/project.yml")
+    #
+    # However, because monorepo, this takes literally minutes to scan the entire repository, which we can't afford to
+    # do every time we run a command on mpyl. For this reason, I decided to go with an optimized find command, which
+    # is a bit harder to read but completes in a couple of seconds.
+    # If you find a faster way to list all files, please raise a PR.
+
+    command = (
```

Comment:
Can we maybe use `os` from python to make it more easy to read and avoid using `subprocess`? Or is that also slow?

Reply:
`subprocess` is also part of the standard library, and in fact the documentation of `os.system` even specifically says:

> The [subprocess](https://docs.python.org/3/library/subprocess.html#module-subprocess) module provides more powerful facilities for spawning new processes and retrieving their results; **using that module is preferable to using this function**.